### ISSUE TITLE
Trailing Line on AreaChart if values aren't defined

### ISSUE THREAD

**Issue by CharlesMcMillan** [recharts/recharts#2176]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://jsfiddle.net/mutualslump/Lwvymzrx/4/](https://jsfiddle.net/mutualslump/Lwvymzrx/4/)

### Steps to reproduce
Fill in an AreaChart with array values for the datakey (lower/upper bounds)...leave some of these datakeys null/undefined in the data array

### What is expected?
A hard cutoff of the area chart, so that it doesn't try to draw a line from the null values to the non null values

### What is actually happening?
When you look at the linked repro (https://jsfiddle.net/mutualslump/Lwvymzrx/4/), you'll see that recharts is drawing a line from 'Page D' (which does not have any areacharts datakey to 'Page E' (which does have that datakey)

| Environment | Info |
|---|---|
| Recharts | v2.0.0-beta.6 |
| React | 15.6.2 |
| System | MacOS |
| Browser | Chrome 83.0 |

**Comment by andriyloza** [comment]
I have the same issue with AreaChart and try to draw line instead of triangles in case null/undefined value

**Comment by hbermani** [comment]
Facing the same issue here.  connectNulls={true} didn't resolve it for me. 

In the end I just used a hacky way to not display the stroke and just display the filled area, I provided stroke="00B1EB" value instead of "#00B1EB" so it doesn't draw it and only draw the area.

**Comment by seiyab** [comment]
Here I reproduced in CodeSandbox.
https://codesandbox.io/s/serene-lichterman-g0s9t6?file=/src/Repro.tsx

**Comment by ckifer** [comment]
Issue happened between these two releases https://github.com/recharts/recharts/compare/v2.0.0-beta.3...v2.0.0-beta.4

culprit looks to be https://github.com/recharts/recharts/compare/v2.0.0-beta.3...v2.0.0-beta.4#diff-1496ed5301e625b1e310643256ed6b2af380dd919af777f25d65beecac8f7b22R195-R206

When going to fix though things get confusing fast. What looks to be an easy fix (account for connectNulls) seems to be that `connectNulls` never really worked as expected and axis.scale(value[1]) (which returns a value) was giving the previous behavior. Will keep digging.

**Comment by ckifer** [comment]
Actually, when looking at the code it is behaving correctly _for the most part_. `connectNulls` is not meant to create 0 point data where there is none. It is meant to bridge lines/areas where between two or more values there is a gap.

In the current docs description
> Whether to connect a graph line across null points.

Which is not the same as if it were "Whether to connect a graph line across null points **and treat nulls as zero.** 

If I [modify the repro a bit](https://codesandbox.io/s/great-easley-h28slj) and add a `null` point in the middle of two existing points then they are bridged via `connectNulls`.
![image](https://github.com/recharts/recharts/assets/25180830/89c9d0d7-a7fb-493a-944b-c5fdc0700948)

At the start and end of my chart though, I have no data. There is nothing to bridge the gap to or from.

_However_ libraries like [Highcharts](https://api.highcharts.com/highcharts/plotOptions.series.connectNulls) treat nulls as 0 only in stacked area charts. We should too. Warrants more digging in `Curve` I believe

**Comment by seiyab** [comment]
Thank you so much for sharing the progress 🚀 
The current behavior might be valid in a sense. However, it is counterintuitive and I think most people should expect stacked area chart to show nulls as 0.
I agree with you. Thanks.

**Comment by ckifer** [comment]
Yeah I'll keep on this. Just a lot of unfamiliar code to go through without much time to do it. Will document my progress here for myself as much as anyone else. Might be a while though

**Comment by HHongSeungWoo** [comment]
This was changed due to https://github.com/recharts/recharts/issues/1601

**Comment by seiyab** [comment]
So just showing undefined as 0 can be considered as regression for https://github.com/recharts/recharts/issues/1601?
Can following behavior acceptable? I haven't make sure it's feasible, though.
- If none of stacked area have value, area is not shown as suggested in https://github.com/recharts/recharts/issues/1601
- Or else (one or more have value), null-like value are shown as 0

I feel that it's hard to find consistent & useful behavior...

If we can't find acceptable default behavior, should we have a new option like `showEmptyAsZero: boolean`?

**Comment by ckifer** [comment]
The definition of reliable and consistent is different for everyone is the problem 😅. For the individual that reported the original issue, that is good behavior, but for you and others it is not. 

I do not want to regress previous fixes. I don't really want to add new props either. 

I think your suggestion is just undoing that PR which would be the regression you mention? What is different?

**Comment by seiyab** [comment]
My suggestion is not just reverting the PR. I draw a figure for my thought.
<img width="451" alt="image" src="https://github.com/recharts/recharts/assets/20365512/1077a5c5-7071-4c2c-9bf5-9bc30923f982">
Can this help you to understand my thought?

(I'm not very sure that I understand https://github.com/recharts/recharts/issues/1601)

**Comment by seiyab** [comment]
I forked jsfiddle of https://github.com/recharts/recharts/issues/1601 to make it work again.
https://jsfiddle.net/tk45fgqL/
<img width="416" alt="image" src="https://github.com/recharts/recharts/assets/20365512/37e59b98-d84a-4670-999e-1d361e1bea1d">
```js
const data = [
      {month: '2015.01', a: 4000, b: 2400, c: 2400},
      {month: '2015.02', a: 3000, b: 1398, c: 2210},
      {month: '2015.03', a: 2000, b: 9800, c: 2290},
      {month: '2015.04', a: 2780, b: 3908, c: 2000},
      {month: '2015.05', a: 1890, b: 4800, c: 2181},
      {month: '2015.06', a: 2390, b: 3800, c: 2500},
      {month: '2015.07', a: 3490, b: 4300, c: 2100},
      {month: '2015.07'}, // the issue points this entry
];
```

**Comment by ckifer** [comment]
Hm that could make sense. The problem is how people interpret it. If it's truly a zero data point should not the data in the data array be zero? That makes this all go away. 

Anyways, I still think we should follow what other charting libraries do - so we can be consistent with something pre existing at least. That might mean doing what you suggest or looking closer at the use case from the other issue and evaluating if we can accomplish a different way. Or a new prop
I guess 😅. Either way I think needs a little more research.

**Comment by seiyab** [comment]
> Anyways, I still think we should follow what other charting libraries do - so we can be consistent with something pre existing at least.

Makes sense and looks a reasonable approach. I might research when I find some time.

**Comment by ckifer** [comment]
Thanks, feel free to document here. This would be good to change in 3.x as breaking if we want to

**Comment by seiyab** [comment]
I'm writing report in this gist.
https://gist.github.com/seiyab/71386e16f16ee9e6d7465ac5ee9e0609

**Comment by seiyab** [comment]
Ah, I had misunderstood https://github.com/recharts/recharts/issues/2176#issuecomment-1594330301.
> However libraries like [Highcharts](https://api.highcharts.com/highcharts/plotOptions.series.connectNulls) treat nulls as 0 only in stacked area charts. We should too. Warrants more digging in Curve I believe

This suggests treat nulls as 0 only in stacked area charts if `connectNulls=true`. I missed the implicit "if statement" 😅 .
Treating nulls as 0 if `connectNulls=true` doesn't look to conflict with #1601 so it should be a reasonable approach. I also think that it doesn't cause practical backward incompatibility. Screenshot pasted in https://github.com/recharts/recharts/issues/2176#issuecomment-1594330301 shows current behavior for stacked area chart + `connectNulls=true` shouldn't be "practically useful", it "technically makes sense", though.
Can't we go same as Highcharts?

---
By the way, researching other libraries, it doesn't look not common problem. For many libraries (and languages), they have *type* so null-like values cause error in compile time or runtime. Additionally, considering `connectNulls`-like feature might not be much common. So I think library that can be reference is rare. Just following HIghcharts you suggested looks reasonable.

**Comment by ckifer** [comment]
Yeah I think that works!

**Comment by PavelVanecek** [comment]
I understand this is now resolved since v3 is released - please leave a comment if there are still some trouble.
